builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -642,8 +673,28 @@ func resourceVSphereVirtualMachineCreate(d *schema.ResourceData, meta interface{
 		for i, v := range vL.([]interface{}) {
 			disk := v.(map[string]interface{})
 			if i == 0 {
-				if v, ok := disk["template"].(string); ok && v != "" {
-					vm.template = v
+				if diskClone, ok := disk["clone"].(map[string]interface{}); ok && len(diskClone) > 0 {
+					vm.clone = &virtualMachineClone{
+						label:  diskClone["label"].(string),
```

Comment:
On the other hand label will always be set if clone is set, since it's a required field. But more checking doesn't hurt.